05/app.js

```diff
@@ -8,19 +8,38 @@ const stats = {
 };
 
 /* tutaj umieść swój kod */
-let counterP = 1;
-let counterA = 1;
+
 const temp = function (e) {
-    if(e.target.tagName == 'A') {
-        e.preventDefault();
-        stats['links']['/dolor.html'] = counterA;
-        counterA++;
+    const href = e.target.getAttribute('href');
+    const par = e.target.dataset.id;
+
+    e.preventDefault();
+
+    if(e.target.tagName == 'A')
+    {
+        if (typeof stats['links'][href] === 'undefined') {
+            stats['links'][href] = 0;
+        }
+        stats['links'][href]++;
+        return
     }
-    else if(e.target.tagName == 'P'){
-        stats['paragraphs']['p1'] = counterP;
-        counterP++;
+
+    if(e.target.tagName == 'P')
+    {
+        if (typeof stats['paragraphs'][par] === 'undefined') {
+            stats['paragraphs'][par] = 0;
+        }
+        stats['paragraphs'][par]++;
+        return
```

Comment:
Zobacz, że ten kod jest bardzo podobny tj.
```
       if (typeof stats['links'][href] === 'undefined') {
            stats['links'][href] = 0;
        }
        stats['links'][href]++;
```

Zmieniają się tylko ta część: `['links'][href]` i `['paragraphs'][par]`, może postarałbyś się utworzyć funkcję, która wykonuje te działąnia? Elementy, które się zmienią należy przekazać przez parametry (każdy element to osobny parametr  więc u nas będą 2)